lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+            disabledTooltip: 'No options',
+            tooltipModifierClass: 'tooltipClass'
+          });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
+
+          expect(tooltip).toHaveTextContent('No options');
+          expect(tooltip.querySelector('.tooltipClass')).toBeInTheDocument();
+        });
+
+        it('does not show when disabledTooltip prop is not provided', async function() {
+          const trigger = renderAndGetTrigger({ disabled: true });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
+          await runTimers();
+
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+        });
+      });
     });
+  });
 
-    it('uses option name as checkbox label', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+  describe('filter', function() {
+    const filterView = (props?: Partial<Props>) => quickRender({
+      ...props,
+      filterThreshold: 2,
+      onFilterChange: jest.fn()
+    });
+
+    it('renders an input with type="text" iff when the number of options ' +
+    'is greater than filterThreshold and onFilterChange is supplied', function() {
+      const noFilterView = quickRender({ onFilterChange: jest.fn() });
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('children', 'Foo');
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('children', 'Bar');
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('children', 'Null');
+      expect(quickRender().queryByRole('textbox')).not.toBeInTheDocument();
+      expect(noFilterView.queryByRole('textbox')).not.toBeInTheDocument();
+      expect(filterView().getByRole('textbox')).toHaveAttribute('type', 'text');
     });
 
-    it('generates checkboxId using name prop and option name', function () {
-      const shallowRender = getShallow();
-      const renderOption = shallowRender.renderProp('renderOption');
+    it('sets the filter input placeholder as specified by filterPlaceholder, ' +
+      'default is "filter"', function() {
+      const viewWithPlaceholder = filterView({ filterPlaceholder: 'filter cats' });
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('checkboxId', 'nx-collapsible-items-select-foobar-foo');
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('checkboxId', 'nx-collapsible-items-select-foobar-bar');
-      expect(renderOption({id: null, name: 'Null'}))
-          .toHaveProp('checkboxId', 'nx-collapsible-items-select-foobar-null');
+      expect(filterView().getByRole('textbox')).toHaveAttribute('placeholder', 'filter');
+      expect(viewWithPlaceholder.getByRole('textbox')).toHaveAttribute('placeholder', 'filter cats');
     });
 
```

Comment:
Needs a test that typing in the filter box triggers `onFilterChange`.

Reply:
Added here 7b25dc023be059b3b34d7bd728f43772aeed5d6f